src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.java

```diff
@@ -167,6 +172,7 @@ public static Container combine(@CheckForNull Container parent, @Nonnull Contain
                 .withEnv(combineEnvVars(parent, template)) //
                 .withEnvFrom(combinedEnvFromSources(parent, template))
                 .withNewSecurityContext().withPrivileged(privileged).endSecurityContext() //
+                .withNewSecurityContext().withRunAsUser(runAsUser).endSecurityContext() //
```

Comment:
Should this not be conditional on `runAsUser` being nonzero?

Reply:
why ? long can be zero , zero mean root user